crates/interpreter/src/value.rs

```diff
@@ -71,3 +71,28 @@ pub fn strict_equality(x: &JSValue, y: &JSValue) -> bool {
         _ => false,
     }
 }
+
+pub fn equality(x: &JSValue, y: &JSValue) -> bool {
+    match (x, y) {
+        (JSValue::Undefined, JSValue::Undefined)
+        | (JSValue::Null, JSValue::Null)
+        | (JSValue::Boolean(_), JSValue::Boolean(_))
+        | (JSValue::Number(_), JSValue::Number(_))
+        | (JSValue::String(_), JSValue::String(_))
+        | (JSValue::Object(_), JSValue::Object(_))
+        | (JSValue::NativeFunction(_), JSValue::NativeFunction(_)) => strict_equality(x, y),
+
+        (JSValue::Undefined, JSValue::Null) => true,
+        (JSValue::Null, JSValue::Undefined) => true,
+        (JSValue::Number(a), JSValue::String(_)) => a == &to_number(y),
+        (JSValue::String(_), JSValue::Number(b)) => &to_number(x) == b,
+        // TODO: If Type(x) is BigInt and Type(y) is String, then ...
+        // TODO: If Type(x) is String and Type(y) is BigInt, return the result of the comparison...
+        (JSValue::Boolean(_), _) => !to_boolean(&JSValue::Number(to_number(x))) == to_boolean(y),
```

Comment:
this doesn't sound follow the spec.
```
Step 8. If Type(x) is Boolean, return the result of the comparison ! ToNumber(x) == y.
```
so it should call `equaliy` with `to_number(x)` and `y`,
and there should not be logical not